src/main/java/seedu/address/logic/commands/AddTeamCommand.java

```diff
@@ -0,0 +1,158 @@
+
+    private final ModuleCode module;
+    private final TutorialClass tutorialClass;
+    private final String teamName;
+    private final int teamSize;
+
+    /**
+     * Constructs an AddTeamCommand to add the specified {@code TutorialTeam} to the
+     * specified {@code TutorialClass}, without the specified team size.
+     * @param module        The module code of the tutorial class to be added.
+     * @param tutorialClass The tutorial class to be added.
+     * @param teamName      The name of the team to be added.
+     */
+    public AddTeamCommand(ModuleCode module, TutorialClass tutorialClass, String teamName) {
+        requireAllNonNull(module);
+        this.module = module;
+        this.tutorialClass = tutorialClass;
+        this.teamName = teamName;
+        this.teamSize = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Constructs an AddTeamCommand to add the specified {@code TutorialTeam} to the
+     * specified {@code TutorialClass}, with the specified team size.
+     * @param module        The module code of the tutorial class to be added.
+     * @param tutorialClass The tutorial class to be added.
+     * @param teamName      The name of the team to be added.
+     * @param teamSize      The size of the team to be added.
+     */
+    public AddTeamCommand(ModuleCode module, TutorialClass tutorialClass, String teamName, int teamSize) {
+        requireAllNonNull(module);
+        this.module = module;
+        this.tutorialClass = tutorialClass;
+        this.teamName = teamName;
+        this.teamSize = teamSize;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        ModuleTutorialPair moduleAndTutorialClass = getModuleAndTutorialClass(model);
+        ModuleCode module = moduleAndTutorialClass.getModule();
+        TutorialClass tutorialClass = moduleAndTutorialClass.getTutorialClass();
+
+        TutorialTeam newTeam = new TutorialTeam(teamName, teamSize);
+
+        if (tutorialClass.hasTeam(newTeam)) {
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_TEAM, teamName, module, tutorialClass));
+        } else {
+            tutorialClass.addTeam(newTeam);
+        }
+
+        if (teamSize != Integer.MAX_VALUE) {
+            return new CommandResult(generateSuccessMessage(module, tutorialClass, teamName, teamSize));
+        } else {
+            return new CommandResult(generateSuccessMessage(module, tutorialClass, teamName));
+        }
+    }
+
+    protected ModuleTutorialPair getModuleAndTutorialClass(Model model) throws CommandException {
+        requireNonNull(model);
+        ModuleCode module = getModule();
+        TutorialClass tutorialClass = getTutorialClass();
+        ModuleCode existingModule = model.findModuleFromList(module);
+        TutorialClass existingTutorialClass = model.findTutorialClassFromList(tutorialClass, existingModule);
```

Comment:
Will the code here throw a null pointer exception if the existingModule == null?

Reply:
Yes, it is handled in the next line. Check out line 98 and 101 if that is what you meant

```
if (existingModule == null) {
            throw new CommandException(String.format(ModuleMessages.MESSAGE_MODULE_NOT_FOUND, module));
        }
        if (existingTutorialClass == null) {
            throw new CommandException(
                    String.format(ModuleMessages.MESSAGE_TUTORIAL_DOES_NOT_BELONG_TO_MODULE, tutorialClass, module));
        }
```

Reply:
My only concern now is that when .findTutorialClassFromList throws a null pointer exception that this exception is not caught in the subsequent execution of the code and may result in bugs.

Reply:
@Zack-Tay it will return null when an exception is thrown 😄 
```
    @Override
    public TutorialClass findTutorialClassFromList(TutorialClass tutorialClass, ModuleCode moduleCode) {
        try {
            return addressBook.findTutorialClassFromList(tutorialClass, moduleCode);
        } catch (Exception e) {
            return null;
        }
    }
```

Reply:
Based on my initial testing, the error has been handled.

<img width="720" alt="Screenshot 2024-03-24 at 10 06 04 PM" src="https://github.com/AY2324S2-CS2103T-T09-4/tp/assets/84561814/1d355b04-7a67-45d3-97ff-ba15b7334660">